src/ptr/ref_count.zig

```diff
@@ -63,8 +63,9 @@ pub const RefCountOptions = struct {
 ///
 ///     const ref_ptr = RefPtr(T).initRef(existing_raw_pointer);
 ///
-pub fn RefCount(T: type, field_name: []const u8, destructor: fn (*T) void, options: RefCountOptions) type {
+pub fn RefCount(T: type, field_name: []const u8, destructor_untyped: anytype, options: RefCountOptions) type {
```

Comment:
why `anytype`?

Reply:
issue in the compiler where analyzing the param type *T will analyze T to find out if it has any comptime fields to determine if the error "destructor must be marked comptime since it is a comptime only type" is going to be present or not.